core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -478,12 +478,11 @@ class TimeSeriesShard(val dataset: Dataset,
       val partId = partIterator.next()
```

Comment:
can we make result generation lazy by wrapping partIterator ? This will reduce memory requirements.